src/category_theory/opposites.lean

```diff
@@ -45,6 +45,9 @@ def unop (X : Cᵒᵖ) : C := X
 
 attribute [irreducible] opposite
 
+def op_induction {F : Π (X : Cᵒᵖ), Sort v₁} (h : Π X, F (op X)) : Π X, F X :=
+λ X, h (unop X)
+
```

Comment:
Sneaky! I like it.

Reply:
I wanted to use it in the form
```lean
  induction U using op_induction,
```
to avoid the whole `revert`/`intro` dance, but there's a bug in `induction` with custom induction principles which produces an ill-typed proof state.